### PR TITLE
[READY] Issue-263 - Updating openwrt and opkg versions

### PR DESCRIPTION
**Pull request by sarcasticadmin**
## Description of PR
Fixes: #263 #119 

Updating openwrt and pkg versions to the latest builds. Incidentally the WNDR hardware had its build targets updated to ath79 as part of this, fullfilling #119 There was also the need to upgrade the docker image to `python3` and a workaround for a file too long for `ipq806x` builds. All and all the hardware looks good and comes up as expected with pretty new openwrt and opkg src trees.

## Previous Behavior
* Previous years version of openwrt
* Previous years version of openwrt pkgs
* `docker` image used `python2`

## New Behavior
* Latest version of openwrt
* Latest version of openwrt pkgs
* Docker image updated to `python3`
* Docker image push to https://hub.docker.com/r/sarcasticadmin/openwrt-build/tags
* Updated circleci docker builds for weekly builds
* Added `image info` to the `serverspec` tests so it outputs version info of hardware where the tests ran. Example:
```
Image info:
Linux OpenWrt 4.19.98 #0 Thu Feb 6 02:51:44 2020 mips GNU/Linux
SCALE_VER=443b7fc8445442735d4eaeb60bb16dd67f60fde9
OPENWRT_VER=87b14bc6c289e7e80052b344c0f0c35e95ced267
BUILD_ID="r0-1f3bfcd"
```
* Updated the documentation in various parts related to `openwrt`
* Worked around `ipq806x` build errors by switch to  short-SHA path names. This was the error I was seeing (will bring it up with upstream openwrt):
```
make[5]: stat: src/abi/libsupc/._home_openwrt_scale-network_openwrt_build_source-ipq806x-deb835849aeb713968aaa48bf046101140ab4825_staging_dir_toolchain-arm_cortex-a15+neon-vfpv4_gcc-8.3.0_musl_eabi_lib_gcc_arm-openwrt-linux-muslgnue
abi_8.3.0_.._.._.._.._arm-openwrt-linux-muslgnueabi_lib_libsupc++.a.dep: File name too long
```
* Marking ipq806x as broken with the current version of upstream that we are tracking
## Tests
* Tests against each version of our hardware (see below)
* Tested each version of the hardware for functionality
* Validated pkg upgrades too (see below)

### WNDR

Output of system pkgs:
```
$ ssh root@192.168.89.129 "opkg list-installed"
apinger - 2015-04-09-78eb3287-1
base-files - 210-r0-1f3bfcd
bash - 5.0-3
busybox - 1.31.1-1
fstools - 2019-11-03-f438552e-1
fwtool - 2019-11-12-8f7fe925-1
getrandom - 2019-06-16-4df34a4d-3
hostapd-common - 2019-08-08-ca8c2bd2-2
ip6tables - 1.8.3-2
iperf3 - 3.7-1
iptables - 1.8.3-2
iw - 5.3-2
iwinfo - 2019-10-16-07315b6f-1
jq - 1.6-1
jshn - 2019-11-24-07413cce-1
jsonfilter - 2018-02-04-c7e938d6-1
kernel - 4.19.86-1-1994196a7a16597b0fe30e2a45c4a58f
kmod-ath - 4.19.86+5.4-rc8-1-1
kmod-ath9k - 4.19.86+5.4-rc8-1-1
kmod-ath9k-common - 4.19.86+5.4-rc8-1-1
kmod-cfg80211 - 4.19.86+5.4-rc8-1-1
kmod-gpio-button-hotplug - 4.19.86-3
kmod-ip6tables - 4.19.86-1
kmod-ipt-conntrack - 4.19.86-1
kmod-ipt-core - 4.19.86-1
kmod-ipt-nat - 4.19.86-1
kmod-ipt-offload - 4.19.86-1
kmod-leds-reset - 4.19.86-1
kmod-lib-crc-ccitt - 4.19.86-1
kmod-mac80211 - 4.19.86+5.4-rc8-1-1
kmod-nf-conntrack - 4.19.86-1
kmod-nf-conntrack6 - 4.19.86-1
kmod-nf-flow - 4.19.86-1
kmod-nf-ipt - 4.19.86-1
kmod-nf-ipt6 - 4.19.86-1
kmod-nf-nat - 4.19.86-1
kmod-nf-reject - 4.19.86-1
kmod-nf-reject6 - 4.19.86-1
kmod-nls-base - 4.19.86-1
kmod-owl-loader - 4.19.86+5.4-rc8-1-1
kmod-phy-ath79-usb - 4.19.86-1
kmod-ppp - 4.19.86-1
kmod-pppoe - 4.19.86-1
kmod-pppox - 4.19.86-1
kmod-slhc - 4.19.86-1
kmod-usb-core - 4.19.86-1
kmod-usb-ehci - 4.19.86-1
kmod-usb-ledtrig-usbport - 4.19.86-1
kmod-usb-ohci - 4.19.86-1
kmod-usb-serial - 4.19.86-1
kmod-usb-serial-ftdi - 4.19.86-1
kmod-usb-serial-pl2303 - 4.19.86-1
kmod-usb2 - 4.19.86-1
libblobmsg-json - 2019-11-24-07413cce-1
libbz2-1.0 - 1.0.8-1
libc - 1.1.24-2
libestr - 0.1.11-2
libevent2-7 - 2.1.11-2
libfastjson - 0.99.8-1
libffi - 3.2.1-3
libgcc1 - 8.3.0-2
libip4tc2 - 1.8.3-2
libip6tc2 - 1.8.3-2
libiwinfo-lua - 2019-10-16-07315b6f-1
libiwinfo20181126 - 2019-10-16-07315b6f-1
libjson-c4 - 0.13.1-1
libjson-script - 2019-11-24-07413cce-1
liblua5.1.5 - 5.1.5-7
libncurses6 - 6.1-5
libnetsnmp - 5.8-1
libnl-tiny - 2019-10-29-0219008c-1
libopenssl1.1 - 1.1.1d-2
libpcap1 - 1.9.1-2
libpci - 3.6.2-4
libpcre - 8.43-2
libpopt0 - 1.16-2
libpthread - 1.1.24-2
libreadline8 - 8.0-1
librrd1 - 1.0.50-2
librt - 1.1.24-2
libubox20170601 - 2019-11-24-07413cce-1
libubus-lua - 2018-10-06-221ce7e7-1
libubus20170705 - 2018-10-06-221ce7e7-1
libuci-lua - 2019-11-14-8dd50da2-3
libuci20130104 - 2019-11-14-8dd50da2-3
libuclient20160123 - 2019-05-30-3b3e368d-1
libuuid1 - 2.34-2
libxtables12 - 1.8.3-2
lldpd - 1.0.3-1
logrotate - 3.15.0-2
lua - 5.1.5-7
mtd - 25
netifd - 2019-11-20-e45b1408-2
odhcp6c - 2019-01-11-e199804b-16
openssh-client - 8.1p1-1
openssh-client-utils - 8.1p1-1
openssh-keygen - 8.1p1-1
openssh-server - 8.1p1-1
openwrt-keyring - 2019-07-25-8080ef34-1
opkg - 2019-06-14-dcbc142e-1
ppp - 2.4.7.git-2019-10-04-2
ppp-mod-pppoe - 2.4.7.git-2019-10-04-2
procd - 2019-11-04-3aa051b4-4
python3-base - 3.8.0-3
python3-light - 3.8.0-3
rrdtool1 - 1.0.50-2
rsyslog - 8.39.0-2
swconfig - 12
tcpdump - 4.9.3-1
terminfo - 6.1-5
uboot-envtools - 2019.07-1
ubox - 2019-06-16-4df34a4d-3
ubus - 2018-10-06-221ce7e7-1
ubusd - 2018-10-06-221ce7e7-1
uci - 2019-11-14-8dd50da2-3
uclient-fetch - 2019-05-30-3b3e368d-1
urandom-seed - 1.0-1
urngd - 2019-10-21-40f939d5-1
usign - 2019-09-21-f34a383e-1
wireless-regdb - 2019.06.03
wpad-basic - 2019-08-08-ca8c2bd2-2
zabbix-agentd - 4.0.11-2
zabbix-extra-mac80211 - 4.0.11-2
zabbix-extra-network - 4.0.11-2
zabbix-extra-wifi - 4.0.11-2
zabbix-sender - 4.0.11-2
zlib - 1.2.11-3
```

`serverspec`
```
$ rake spec TEST_TYPE=openwrt TARGET_HOST=192.168.89.129
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb

Image info:
Linux OpenWrt 4.19.86 #0 Fri Jan 10 06:48:20 2020 mips GNU/Linux
SCALE_VER=546889e009e94db9c6b3e177aab82c0d3155720c
OPENWRT_VER=deb835849aeb713968aaa48bf046101140ab4825
BUILD_ID="r0-1f3bfcd"

shared
  Command "which apinger 2> /dev/null"
    exit_status
      should eq 0
  Command "which bash 2> /dev/null"
    exit_status
      should eq 0
  Command "which logrotate 2> /dev/null"
    exit_status
      should eq 0
  Command "which python3 2> /dev/null"
    exit_status
      should eq 0
  Command "which rsyslogd 2> /dev/null"
    exit_status
      should eq 0
  Command "which zabbix_agentd 2> /dev/null"
    exit_status
      should eq 0
  Command "which tcpdump 2> /dev/null"
    exit_status
      should eq 0
  Command "which snmpd 2> /dev/null"
    exit_status
      should eq 1
  Command "which dropbear 2> /dev/null"
    exit_status
      should eq 1
  Command "pgrep apinger"
    exit_status
      should eq 0
  Command "pgrep crond"
    exit_status
      should eq 0
  Command "pgrep rsyslogd"
    exit_status
      should eq 0
  Command "pgrep zabbix"
    exit_status
      should eq 0
  Command "rsyslogd -N1"
    exit_status
      should eq 0
  File "/root/bin/wifi-details.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  File "/root/bin/config-version.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  Command "/root/bin/config-version.sh"
    exit_status
      should eq 0
  Command "/root/bin/config-version.sh -c 9999"
    exit_status
      should eq 1
  File "/etc/scale-release"
    should exist
    should be mode 644
    should be owned by "root"
    should be grouped into "root"
  Command "source /etc/scale-release && test -z $SCALE_VER"
    exit_status
      should eq 1
  Command "source /etc/scale-release && test -z $OPENWRT_VER"
    exit_status
      should eq 1
  File "/etc/resolv.conf"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/network"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/wireless"
    should be symlink
    should be owned by "root"
    should be grouped into "root"

Finished in 3.2 seconds (files took 5.01 seconds to load)
39 examples, 0 failures
```

## TPLINK

`pkgs`
```
ssh root@192.168.89.131 "opkg list-installed"
apinger - 2015-04-09-78eb3287-1
ath10k-firmware-qca99x0-ct - 2019-10-03-d622d160-1
base-files - 210-r0-1f3bfcd
bash - 5.0-3
busybox - 1.31.1-1
fstools - 2019-11-03-f438552e-1
fwtool - 2019-11-12-8f7fe925-1
getrandom - 2019-06-16-4df34a4d-3
hostapd-common - 2019-08-08-ca8c2bd2-2
ip6tables - 1.8.3-2
iperf3 - 3.7-1
iptables - 1.8.3-2
iw - 5.3-2
iwinfo - 2019-10-16-07315b6f-1
jq - 1.6-1
jshn - 2019-11-24-07413cce-1
jsonfilter - 2018-02-04-c7e938d6-1
kernel - 4.14.156-1-93989859147147dff37ee71571175c37
kmod-ata-ahci - 4.14.156-1
kmod-ata-ahci-platform - 4.14.156-1
kmod-ata-core - 4.14.156-1
kmod-ath - 4.14.156+5.4-rc8-1-1
kmod-ath10k-ct - 4.14.156+2019-09-09-5e8cd86f-1
kmod-cfg80211 - 4.14.156+5.4-rc8-1-1
kmod-gpio-button-hotplug - 4.14.156-3
kmod-hwmon-core - 4.14.156-1
kmod-ip6tables - 4.14.156-1
kmod-ipt-conntrack - 4.14.156-1
kmod-ipt-core - 4.14.156-1
kmod-ipt-nat - 4.14.156-1
kmod-ipt-offload - 4.14.156-1
kmod-leds-gpio - 4.14.156-1
kmod-lib-crc-ccitt - 4.14.156-1
kmod-mac80211 - 4.14.156+5.4-rc8-1-1
kmod-nf-conntrack - 4.14.156-1
kmod-nf-conntrack6 - 4.14.156-1
kmod-nf-flow - 4.14.156-1
kmod-nf-ipt - 4.14.156-1
kmod-nf-ipt6 - 4.14.156-1
kmod-nf-nat - 4.14.156-1
kmod-nf-reject - 4.14.156-1
kmod-nf-reject6 - 4.14.156-1
kmod-nls-base - 4.14.156-1
kmod-ppp - 4.14.156-1
kmod-pppoe - 4.14.156-1
kmod-pppox - 4.14.156-1
kmod-scsi-core - 4.14.156-1
kmod-slhc - 4.14.156-1
kmod-usb-core - 4.14.156-1
kmod-usb-dwc3 - 4.14.156-1
kmod-usb-dwc3-of-simple - 4.14.156-1
kmod-usb-ehci - 4.14.156-1
kmod-usb-ledtrig-usbport - 4.14.156-1
kmod-usb-ohci - 4.14.156-1
kmod-usb-phy-qcom-dwc3 - 4.14.156-1
kmod-usb-serial - 4.14.156-1
kmod-usb-serial-ftdi - 4.14.156-1
kmod-usb-serial-pl2303 - 4.14.156-1
kmod-usb2 - 4.14.156-1
kmod-usb3 - 4.14.156-1
libblobmsg-json - 2019-11-24-07413cce-1
libbz2-1.0 - 1.0.8-1
libc - 1.1.24-2
libestr - 0.1.11-2
libevent2-7 - 2.1.11-2
libfastjson - 0.99.8-1
libffi - 3.2.1-3
libgcc1 - 8.3.0-2
libip4tc2 - 1.8.3-2
libip6tc2 - 1.8.3-2
libiwinfo-lua - 2019-10-16-07315b6f-1
libiwinfo20181126 - 2019-10-16-07315b6f-1
libjson-c4 - 0.13.1-1
libjson-script - 2019-11-24-07413cce-1
liblua5.1.5 - 5.1.5-7
libncurses6 - 6.1-5
libnetsnmp - 5.8-1
libnl-tiny - 2019-10-29-0219008c-1
libopenssl1.1 - 1.1.1d-2
libpcap1 - 1.9.1-2
libpci - 3.6.2-4
libpcre - 8.43-2
libpopt0 - 1.16-2
libpthread - 1.1.24-2
libreadline8 - 8.0-1
librrd1 - 1.0.50-2
librt - 1.1.24-2
libubox20170601 - 2019-11-24-07413cce-1
libubus-lua - 2018-10-06-221ce7e7-1
libubus20170705 - 2018-10-06-221ce7e7-1
libuci-lua - 2019-11-14-8dd50da2-3
libuci20130104 - 2019-11-14-8dd50da2-3
libuclient20160123 - 2019-05-30-3b3e368d-1
libuuid1 - 2.34-2
libxtables12 - 1.8.3-2
lldpd - 1.0.3-1
logrotate - 3.15.0-2
lua - 5.1.5-7
mtd - 25
netifd - 2019-11-20-e45b1408-2
odhcp6c - 2019-01-11-e199804b-16
openssh-client - 8.1p1-1
openssh-client-utils - 8.1p1-1
openssh-keygen - 8.1p1-1
openssh-server - 8.1p1-1
openwrt-keyring - 2019-07-25-8080ef34-1
opkg - 2019-06-14-dcbc142e-1
ppp - 2.4.7.git-2019-10-04-2
ppp-mod-pppoe - 2.4.7.git-2019-10-04-2
procd - 2019-11-04-3aa051b4-4
python3-base - 3.8.0-3
python3-light - 3.8.0-3
rrdtool1 - 1.0.50-2
rsyslog - 8.39.0-2
swconfig - 12
tcpdump - 4.9.3-1
terminfo - 6.1-5
ubi-utils - 2.1.1-1
uboot-envtools - 2019.07-1
ubox - 2019-06-16-4df34a4d-3
ubus - 2018-10-06-221ce7e7-1
ubusd - 2018-10-06-221ce7e7-1
uci - 2019-11-14-8dd50da2-3
uclient-fetch - 2019-05-30-3b3e368d-1
urandom-seed - 1.0-1
urngd - 2019-10-21-40f939d5-1
usign - 2019-09-21-f34a383e-1
wireless-regdb - 2019.06.03
wpad-basic - 2019-08-08-ca8c2bd2-2
zabbix-agentd - 4.0.11-2
zabbix-extra-mac80211 - 4.0.11-2
zabbix-extra-network - 4.0.11-2
zabbix-extra-wifi - 4.0.11-2
zabbix-sender - 4.0.11-2
zlib - 1.2.11-3
```
> NOTE: ath10k-firmware-qca99x0-ct - 2019-10-03-d622d160-1
> series of radio firmware

`serverspec`:
```
$ rake spec TEST_TYPE=openwrt TARGET_HOST=192.168.89.131
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb

Image info:
Linux OpenWrt 4.14.156 #0 SMP Fri Jan 10 06:45:50 2020 armv7l GNU/Linux
SCALE_VER=4b167996477d7d175754342d4b304041945638cb
OPENWRT_VER=deb835849aeb713968aaa48bf046101140ab4825
BUILD_ID="r0-1f3bfcd"
OPENWRT_BOARD="ipq806x/generic"
OPENWRT_ARCH="arm_cortex-a15_neon-vfpv4"

shared
  Command "which apinger 2> /dev/null"
    exit_status
      should eq 0
  Command "which bash 2> /dev/null"
    exit_status
      should eq 0
  Command "which logrotate 2> /dev/null"
    exit_status
      should eq 0
  Command "which python3 2> /dev/null"
    exit_status
      should eq 0
  Command "which rsyslogd 2> /dev/null"
    exit_status
      should eq 0
  Command "which zabbix_agentd 2> /dev/null"
    exit_status
      should eq 0
  Command "which tcpdump 2> /dev/null"
    exit_status
      should eq 0
  Command "which snmpd 2> /dev/null"
    exit_status
      should eq 1
  Command "which dropbear 2> /dev/null"
    exit_status
      should eq 1
  Command "pgrep apinger"
    exit_status
      should eq 0
  Command "pgrep crond"
    exit_status
      should eq 0
  Command "pgrep rsyslogd"
    exit_status
      should eq 0
  Command "pgrep zabbix"
    exit_status
      should eq 0
  Command "rsyslogd -N1"
    exit_status
      should eq 0
  File "/root/bin/wifi-details.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  File "/root/bin/config-version.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  Command "/root/bin/config-version.sh"
    exit_status
      should eq 0
  Command "/root/bin/config-version.sh -c 9999"
    exit_status
      should eq 1
  File "/etc/scale-release"
    should exist
    should be mode 644
    should be owned by "root"
    should be grouped into "root"
  Command "source /etc/scale-release && test -z $SCALE_VER"
    exit_status
      should eq 1
  Command "source /etc/scale-release && test -z $OPENWRT_VER"
    exit_status
      should eq 1
  File "/etc/resolv.conf"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/network"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/wireless"
    should be symlink
    should be owned by "root"
    should be grouped into "root"

Finished in 3.9 seconds (files took 4.28 seconds to load)
39 examples, 0 failures
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,21 +16,7 @@
...
 
-### Installation
-#### Gomplate
-Installation of `gomplate` is a little bit tricky since it doesnt come in a `.deb`:
-```bash
-sudo -i
-cd /usr/local/bin/
-curl -O https://github.com/hairyhenderson/gomplate/releases/download/<version>/gomplate_linux-amd64 -L
-mv gomplate_linux-amd64 gomplate
-```
```
Comment (kylerisse) [inline README.md]: Much cleaner now!

File: openwrt/AR71XX.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+cd /tmp
+sysupgrade -v <sysupgrade.bin> # Wait for the AP to load and reboot
+```
+
+Once it comes back online `ssh` back in an confirm the version
+
+```sh
+ssh root@<AP IP>
+cat /etc/os-release
+```
```
Comment (kylerisse) [inline openwrt/AR71XX.md]: Thanks for clear instructions, I will need these!

File: openwrt/Dockerfile [hunk 3]
```diff
@@ -4,7 +4,7 @@ FROM ubuntu:16.04 as build
 RUN apt-get update &&\
     apt-get install -y sudo time git-core subversion build-essential gcc-multilib \
                        libncurses5-dev zlib1g-dev gawk flex gettext wget unzip \
-                       curl python &&\
+                       curl python3 &&\
```
Comment (kylerisse) [inline openwrt/Dockerfile]: best python!

File: openwrt/Makefile [hunk 4]
```diff
@@ -69,6 +70,9 @@ targetconfig: diffconfig
 
 config: $(BUILD_DIR)/$(IMAGEBUILDER)/.config
 
+menuconfig:
+	@cd $(BUILD_DIR)/$(IMAGEBUILDER) && $(MAKE) menuconfig
+
```
Comment (kylerisse) [inline openwrt/Makefile]: very cool

**Comment by sarcasticadmin** [comment]
Turns out the `ipq806x` build is actually having issues with the ath10k drivers on the c2600 :disappointed: It seems like it just took a little while to show up. Im going to look into other solutions but looks like im not the only one: https://bugs.openwrt.org/index.php?do=details&task_id=2614&pagenum=2 for now lets hold off on merging this until I know more

**Comment by sarcasticadmin** [comment]
This also needs to be rebased to have the new required tests.

Update regarding `ipq806x` radio firmware: I have a new build from `openwrt` master that Im currently testing with a few wifi clients. These newer builds includes a fix from late december that could resolve our issue. I should have results from the tests by end of tomorrow 2/11 since thats when the stability issues came up last time on the previous build.

**Comment by sarcasticadmin** [comment]
@owendelong @kylerisse @davidelang 

This is ready to go! I made some notes about the c2600 hardware being broken on upstream openwrt and for `SCaLE 18x`. Well track more progress in #281 . Below are all tests on a Netgear 3800ch on the latest version of openwrt that we are tracking.

## Tests
Currently running on the access point and anymore builds to dogfood before the show but its working well and seems to be stable:
```
BusyBox v1.31.1 () built-in shell (ash)

  _______                     ________        __
 |       |.-----.-----.-----.|  |  |  |.----.|  |_
 |   -   ||  _  |  -__|     ||  |  |  ||   _||   _|
 |_______||   __|_____|__|__||________||__|  |____|
          |__| W I R E L E S S   F R E E D O M
 -----------------------------------------------------
 OpenWrt SNAPSHOT, r0-1f3bfcd
 -----------------------------------------------------
root@OpenWrt:~# uptime
 18:36:31 up 3 days, 15:29,  load average: 0.13, 0.21, 0.22
```
Here is the output of the latest `serverspec`:

```
/usr/home/rherna/.rbenv/versions/2.5.1/bin/ruby -I/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-support-3.8.0/lib:/usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/lib /usr/home/rherna/.rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/openwrt/\*_spec.rb

Image info:
Linux OpenWrt 4.19.98 #0 Thu Feb 6 02:51:44 2020 mips GNU/Linux
SCALE_VER=443b7fc8445442735d4eaeb60bb16dd67f60fde9
OPENWRT_VER=87b14bc6c289e7e80052b344c0f0c35e95ced267
BUILD_ID="r0-1f3bfcd"
OPENWRT_BOARD="ath79/generic"
OPENWRT_ARCH="mips_24kc"

shared
  Command "which apinger 2> /dev/null"
    exit_status
      should eq 0
  Command "which bash 2> /dev/null"
    exit_status
      should eq 0
  Command "which logrotate 2> /dev/null"
    exit_status
      should eq 0
  Command "which python3 2> /dev/null"
    exit_status
      should eq 0
  Command "which rsyslogd 2> /dev/null"
    exit_status
      should eq 0
  Command "which zabbix_agentd 2> /dev/null"
    exit_status
      should eq 0
  Command "which tcpdump 2> /dev/null"
    exit_status
      should eq 0
  Command "which snmpd 2> /dev/null"
    exit_status
      should eq 1
  Command "which dropbear 2> /dev/null"
    exit_status
      should eq 1
  Command "pgrep apinger"
    exit_status
      should eq 0
  Command "pgrep crond"
    exit_status
      should eq 0
  Command "pgrep rsyslogd"
    exit_status
      should eq 0
  Command "pgrep zabbix"
    exit_status
      should eq 0
  Command "rsyslogd -N1"
    exit_status
      should eq 0
  File "/root/bin/wifi-details.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  File "/root/bin/config-version.sh"
    should exist
    should be mode 750
    should be owned by "root"
    should be grouped into "root"
  Command "/root/bin/config-version.sh"
    exit_status
      should eq 0
  Command "/root/bin/config-version.sh -c 9999"
    exit_status
      should eq 1
  File "/etc/scale-release"
    should exist
    should be mode 644
    should be owned by "root"
    should be grouped into "root"
  Command "source /etc/scale-release && test -z $SCALE_VER"
    exit_status
      should eq 1
  Command "source /etc/scale-release && test -z $OPENWRT_VER"
    exit_status
      should eq 1
  File "/etc/resolv.conf"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/network"
    should be symlink
    should be owned by "root"
    should be grouped into "root"
  File "/etc/config/wireless"
    should be symlink
    should be owned by "root"
    should be grouped into "root"

Finished in 7.91 seconds (files took 5.01 seconds to load)
39 examples, 0 failures
```

**Comment by sarcasticadmin** [comment]
Im going to merge this in tomorrow by noon so we can have the image ready for the workparty on saturday, if anyone has any issues with this PR please let me know

**Pull request merged**